sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java

```diff
@@ -0,0 +1,269 @@
+        return this;
+    }
+
+    /**
+     * Gets the idle channel timeout
+     * @return idle channel timeout
+     */
+    public Duration getIdleChannelTimeout() {
+        return idleChannelTimeout;
+    }
+
+    /**
+     * Sets the idle channel timeout
+     * @param idleChannelTimeout idle channel timeout
+     * @return the {@link DirectConnectionConfig}
+     */
+    public DirectConnectionConfig setIdleChannelTimeout(Duration idleChannelTimeout) {
+        this.idleChannelTimeout = idleChannelTimeout;
+        return this;
+    }
+
+    /**
+     * Gets the idle endpoint timeout
+     * @return the idle endpoint timeout
+     */
+    public Duration getIdleEndpointTimeout() {
+        return idleEndpointTimeout;
+    }
+
+    /**
+     * Sets the idle endpoint timeout
+     * @param idleEndpointTimeout the idle endpoint timeout
+     * @return the {@link DirectConnectionConfig}
+     */
+    public DirectConnectionConfig setIdleEndpointTimeout(Duration idleEndpointTimeout) {
+        this.idleEndpointTimeout = idleEndpointTimeout;
+        return this;
+    }
+
+    /**
+     * Gets the max channels per endpoint
+     * @return the max channels per endpoint
+     */
+    public int getMaxChannelsPerEndpoint() {
+        return maxChannelsPerEndpoint;
+    }
+
+    /**
+     * Sets the max channels per endpoint
+     * @param maxChannelsPerEndpoint the max channels per endpoint
+     * @return the {@link DirectConnectionConfig}
+     */
+    public DirectConnectionConfig setMaxChannelsPerEndpoint(int maxChannelsPerEndpoint) {
+        this.maxChannelsPerEndpoint = maxChannelsPerEndpoint;
+        return this;
+    }
+
+    /**
+     * Gets the max requests per endpoint
+     * @return the max requests per endpoint
+     */
+    public int getMaxRequestsPerChannel() {
+        return maxRequestsPerChannel;
+    }
+
+    /**
+     * Sets the max requests per endpoint
+     * @param maxRequestsPerChannel the max requests per endpoint
+     * @return the {@link DirectConnectionConfig}
+     */
+    public DirectConnectionConfig setMaxRequestsPerChannel(int maxRequestsPerChannel) {
+        this.maxRequestsPerChannel = maxRequestsPerChannel;
+        return this;
+    }
+
+    /**
+     * Sets the retry policy options associated with the DocumentClient instance.
+     * <p>
+     * Properties in the RetryOptions class allow application to customize the built-in
+     * retry policies. This property is optional. When it's not set, the SDK uses the
+     * default values for configuring the retry policies.  See RetryOptions class for
+     * more details.
+     *
+     * @param throttlingRetryOptions the RetryOptions instance.
+     * @return the {@link DirectConnectionConfig}.
+     * @throws IllegalArgumentException thrown if an error occurs
+     */
+    @Override
+    public DirectConnectionConfig setThrottlingRetryOptions(ThrottlingRetryOptions throttlingRetryOptions) {
+        super.setThrottlingRetryOptions(throttlingRetryOptions);
```

Comment:
why do we need to overwrite these ones? this is common for both gw and direct.

Reply:
Yes, they are common, and they both call their parent's class APIs.. 
The reason for overriding these ones is because of the return type. 
Since our APIs are fluent type, if you directly use parent's API (which returns `ConnectionConfig`), end users will not be able to use any APIs of `DirectConnectionConfig` or `GatewayConnectionConfig` types..